frontend/src/components/program/DialogActivityCreate.vue

```diff
@@ -61,15 +198,94 @@ export default {
               deleted: false,
             },
           ],
+          copyActivitySource: null,
         })
       } else {
+        this.canReadClipboard = 'unknown'
         // clear the variable parts of the form on exit
+        this.copyActivitySource = null
+        this.copyActivitySourceUrl = null
         this.entityData.location = ''
         this.entityData.scheduleEntries = []
       }
     },
+    copyActivitySourceUrl: function (url) {
+      this.copyActivitySourceUrlLoading = true
+
+      this.getCopyActivitySource(url).then(
+        (activityProxy) => {
+          if (activityProxy != null) {
+            activityProxy._meta.load.then(
+              (activity) => {
+                this.$set(this, 'copyActivitySource', activity)
+                this.$set(this, 'copyContent', activity != null)
+                this.copyActivitySourceUrlLoading = false
+              },
+              () => {
+                this.copyActivitySourceUrlLoading = false
+              }
+            )
+          } else {
+            this.$set(this, 'copyActivitySource', null)
+            this.$set(this, 'copyContent', false)
+            this.copyActivitySourceUrlLoading = false
+          }
+
+          // if Paste-Popover is shown, close it now
+          if (this.copyActivitySourceUrlShowPopover) {
+            this.$nextTick(() => {
+              this.$set(this, 'copyActivitySourceUrlShowPopover', false)
+            })
+          }
+        },
+        () => {
+          this.copyActivitySourceUrlLoading = false
+        }
+      )
+    },
   },
   methods: {
+    refreshCopyActivitySource() {
+      navigator.permissions.query({ name: 'clipboard-read' }).then(
+        (p) => {
+          this.$set(this, 'clipboardPermission', p.state)
+          this.$set(this, 'copyActivitySource', null)
+
+          if (p.state == 'granted') {
+            navigator.clipboard.readText().then((url) => {
+              this.getCopyActivitySource(url).then((activityProxy) => {
+                activityProxy._meta.load.then((activity) =>
+                  this.$set(this, 'copyActivitySource', activity)
```

Comment:
minor:
the experience is not the same if the user gave clipboard access or if he did not.
If you detect an activity in the clipboard, the checkbox to copy the the activity content is not set.
If the user manually pastes the url, the checkbox to copy the activity content is set.

Is this on purpose?